barretenberg/cpp/src/barretenberg/stdlib/primitives/group/cycle_group.cpp

```diff
@@ -59,9 +59,16 @@ cycle_group<Builder>::cycle_group(field_t _x, field_t _y, bool_t is_infinity, bo
         *this = constant_infinity(this->context);
     }
 
-    // We don't support points with only one constant coordinate since valid use-cases are limited and it complicates
-    // the logic
-    BB_ASSERT(_x.is_constant() == _y.is_constant(), "cycle_group: Inconsistent constancy of coordinates");
+    // For the simplicity of methods in this class, we ensure that the coordinates of a point always have the same
+    // constancy. If they don't, we convert the non-constant coordinate to a fixed witness.
+    if (_x.is_constant() != _y.is_constant()) {
+        info("Warning: cycle_group constructed with inconsistent coordinate constancy - converting to fixed witness");
```

Comment:
I think we should remove this warning. If cycle_group does not handle mixed 'constancy', then it's his problem and it should not warn about it because it has a workaround.
This makes the blackbox interface in Noir cleaner and removes complexity by not having to take care of constant/witness inputs to the blackbox. Furthermore, 'DSL' can introduce this inconsistency by itself and getting such warning depending on the inputs looks strange.